pulser/simulation/simulation.py

```diff
@@ -460,6 +462,12 @@ def write_samples(
             )
             samples_dict["phase"][slot.ti : slot.tf] += _pulse.phase
 
+            # Final time-step: repeat last sample
+            if slot.tf == self._tot_duration - 1:
+                samples_dict["amp"][-1] = _pulse.amplitude.samples[-1]
```

Comment:
I think you should put 0 in the amplitude (so actually, not touch it)

Reply:
Removed copying of last value so it's now by default 0.